File: 232.ts
class MyQueue {
    s: Array<number> = [];

    constructor() {
    }

    push(x: number): void {
        console.log(`Pushing ${x}`);
        let st: Array<number> = [];
        let numOps = this.s.length;
        for(let i = 0; i < numOps; i++)
        {
            let tempNum = this.s.pop();
            if(typeof tempNum === 'number')
            {
                st.push(tempNum);
            }
        }

        this.s.push(x);

        for(let j = 0; j < numOps; j++)
        {
            let tempNum = st.pop();
            if(typeof tempNum === 'number')
            {
                this.s.push(tempNum);
            }
        }

        console.log(`After pushing ${x}, s is ${this.s}`);
    }

    pop(): number | undefined {
        if(this.s.length > 0)
        {
            return this.s.splice(-1)[0];
        }
        else
        {
            return undefined;
        }
    }

    peek(): number | undefined {
        if(this.s.length > 0)
        {
            return this.s[this.s.length - 1];
        }
        else
        {
            return undefined;
        }
    }

    empty(): boolean {
        return (this.s.length == 0) ? true : false;
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * var obj = new MyQueue()
 * obj.push(x)
 * var param_2 = obj.pop()
 * var param_3 = obj.peek()
 * var param_4 = obj.empty()
 */

File: 232.py
from collections import deque

class MyQueue:

    def __init__(self):
        self.d = deque()

    def push(self, x: int) -> None:
        dt = deque()
        for i in range(0, len(self.d)):
            dt.append(self.d.pop())

        self.d.append(x)
        for i in range(0, len(dt)):
            self.d.append(dt.pop())     

    def pop(self) -> int:
        return self.d.pop() if len(self.d) > 0 else None        

    def peek(self) -> int:
        return self.d[-1] if len(self.d) > 0 else None

    def empty(self) -> bool:
        return len(self.d) == 0

# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()